api/src/controllers/team.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * This module contains handlers for the "team" route.
+ */
+const axios = require("axios");
+
+const mongoose = require("mongoose");
+
+const Team = require("../models/team.model");
+
+
+const options = {
+    headers: {'Authorization': "Bearer " + process.env.TWITTER_BEARER_TOKEN}
+};
+
+/**
+ * Associates a twitter handle with a team on the /team/twitter-handle/:team-id endpoint. 
+ * @param {*} req request object, containing the team_id in the url and twitter handle in the body
+ * @param {*} res response object
+ * @returns 200: successful added twitter handle to team
+ * @returns 400: team id is not in a valid hexadecimal format
+ * @returns 404: team is not found, or handle is invalid
+ * @returns 500: error trying to update the document in db
+ */
+async function storeHandle(req, res) {
+    const {team_id: team_id} = req.params;
+    const {twitterHandle: handle} = req.body;
+
+    // validate if the team exists
+    if (mongoose.Types.ObjectId.isValid(team_id)) {
+        var foundTeam = await Team.findById(team_id);
+        if (foundTeam == null) {
+            return res.status(404).send(`Error: No team found with given id.`);
+        }
+    } else {
+        return res.status(400).send("Error: Given team id is not in a valid hexadecimal format.");
+    }
+
+    if (handle.length == 0) {  // remove the handle from the doc
+        foundTeam.twitterHandle = undefined
+    } else {  // update the handle
+        // validate the handle by getting user id
+        var response = await axios.get("https://api.twitter.com/2/users/by/username/" + handle, options)
+        if (response.data.errors) {
+            return res.status(400).send("Error: " + response.data.errors[0].detail);
+        } else {
+            foundTeam.twitterHandle = handle;
+            console.log(foundTeam);
+        }
+    }
+
+    try {  // update in db
+        foundTeam.save();
+        return res.status(200).json(foundTeam);
+    } catch (err) {
+        return res.status(500).send(`Error: ${err.message}`);
+    }
+    
+}
+
+/**
+ * Gets the team document from the database on /team/:team_id.
+ * @param {*} req request object, containing team id in the url 
+ * @param {*} res response object, the found team document
+ * @returns 200: the team was found
+ * @returns 404: team is not found
+ * @returns 400: team id is not in a valid hexadecimal format
+ */
+async function getTeam(req, res) {
+    const {team_id: team_id} = req.params;
+
+    // validate if the team exists
+    if (mongoose.Types.ObjectId.isValid(team_id)) {
+        var foundTeam = await Team.findById(team_id);
+        if (foundTeam == null) {
+            return res.status(404).send(`Error: No team found with given id.`);
+        }
+    } else {
+        return res.status(400).send("Error: Given team id is not in a valid hexadecimal format.");
+    }
```

Comment:
for refactoring this duplicated block, I'm thinking we can use middleware.
the flow I'm thinking of: 
```
request -> validate mongo object id middleware function -> if success, validate team exists middleware function -> if success, `next()` into the corresponding handler.
```
Similar to what we're doing for the publications: 
https://github.com/Researchify/Researchify/blob/main/api/src/routes/publications.js#L14
https://github.com/Researchify/Researchify/blob/main/api/src/middleware/publications.js

Though this may be tedious, and I'm okay with leaving as is as it's just a couple of lines.